Sources/Yams/YamlError.swift

```diff
@@ -81,6 +81,13 @@ public enum YamlError: Swift.Error {
             return text + " in line \(mark.line), column \(mark.column)\n"
         }
     }
+
+#if swift(>=4.1.50)
+    @available(*, unavailable, renamed: "reader(problem:offset:value:yaml:)")
+    public static func reader(problem: String, byteOffset: Int, value: Int32, yaml: String) {
```

Comment:
Should we return `YamlError` here to mimic a case member? And should we proxy this to the new member? Honest questions.

```swift
public static func reader(problem: String, byteOffset: Int, value: Int32, yaml: String) -> YamlError {
    return .reader(problem: problem, offset: byteOffset, value: value, yaml: yaml)
}
```

Reply:
> Should we return `YamlError` here to mimic a case member?

Calculating `offset` from `byteOffset` requires parser's encoding, but it is unknown at here.
So we can't return it.

Before this PR, parser's encoding was determined at compile time, so `byteOffset` could be converted to `line` and `column` in yaml string on generating `YamlError.description` later.
This PR changes `YamlError.init(from:with:)` to converting byte offset into `String.IndexDistance?` with using `parser`'s encoding.

Reply:
👍 thanks for explaining!